cpp/include/cudf/io/orc_metadata.hpp

```diff
@@ -25,31 +25,236 @@
+ * contains one element per stripe, where each element contains column statistics for each column.
+ */
+struct raw_orc_statistics {
+  std::vector<std::string> column_names;
+  std::vector<std::string> file_stats;
+  std::vector<std::vector<std::string>> stripes_stats;
+};
+
+/**
+ * @brief Reads file-level and stripe-level statistics of ORC dataset.
  *
  * @ingroup io_readers
  *
  * The following code snippet demonstrates how to read statistics of a dataset
  * from a file:
  * @code
- *  std::string filepath = "dataset.orc";
- *  auto result = cudf::read_orc_statistics(cudf::source_info(filepath));
+ *  auto result = cudf::read_raw_orc_statistics(cudf::source_info("dataset.orc"));
  * @endcode
  *
  * @param src_info Dataset source
  *
- * @return Decompressed ColumnStatistics blobs stored in a vector of vectors. The first element of
- * result vector, which is itself a vector, contains the name of each column. The second element
- * contains statistics of each column of the whole file. Remaining elements contain statistics of
- * each column for each stripe.
+ * @return Column names and encoded ORC statistics
+ */
+raw_orc_statistics read_raw_orc_statistics(source_info const& src_info);
+
+/**
+ * @brief Enumerator for types of column statistics that can be included in `column_statistics`.
+ *
+ * The statistics type depends on the column data type.
+ */
+enum class statistics_type {
+  NONE,
+  INT,
+  DOUBLE,
+  STRING,
+  BUCKET,
+  DECIMAL,
+  DATE,
+  BINARY,
+  TIMESTAMP,
+};
+
+/**
+ * @brief Base class for column statistics that include optional minimum and maximum.
+ *
+ * Includes accessors for the minimum and maximum values.
+ */
+template <typename T>
+struct minmax_statistics {
+  std::unique_ptr<T> _minimum;
+  std::unique_ptr<T> _maximum;
+
+  auto has_minimum() const { return _minimum != nullptr; }
+  auto has_maximum() const { return _maximum != nullptr; }
+  auto minimum() const { return _minimum.get(); }
+  auto maximum() const { return _maximum.get(); }
+};
+
+/**
+ * @brief Base class for column statistics that include an optional sum.
+ *
+ * Includes accessors for the sum value.
+ */
+template <typename T>
+struct sum_statistics {
+  std::unique_ptr<T> _sum;
+
+  auto has_sum() const { return _sum != nullptr; }
+  auto sum() const { return _sum.get(); }
+};
+
+/**
+ * @brief Statistics for integral columns.
+ */
+struct integer_statistics : minmax_statistics<int64_t>, sum_statistics<int64_t> {
+  static constexpr statistics_type type = statistics_type::INT;
+};
+
+/**
+ * @brief Statistics for floating point columns.
+ */
+struct double_statistics : minmax_statistics<double>, sum_statistics<double> {
+  static constexpr statistics_type type = statistics_type::DOUBLE;
+};
+
+/**
+ * @brief Statistics for string columns.
+ *
+ * The `minimum` and `maximum` are the first and last element in hexicographical order
+ * (respectively). The `sum` is the total length of elements in the column.
```

Comment:
```suggestion
 * The `minimum` and `maximum` are the first and last elements, respectively, in lexicographical order.
 * The `sum` is the total length of elements in the column.
```

Is it the total number of characters in all strings in the column? Or the total number of strings in the column?  Comment is unclear.